conda/env/installers/conda.py

```diff
@@ -43,8 +59,35 @@ def dry_run(specs, args, env, *_, **kwargs):
     return solved_env
 
 
-def install(prefix, specs, args, env, *_, **kwargs):
-    """Install packages into an environment"""
+def install(prefix: str, specs: list, args, env, *_, **kwargs) -> dict:
+    """Install packages into a conda environment.
+
+    This function handles two main paths:
+    1. For ExplicitEnvironment instances (from @EXPLICIT files): Bypasses the solver
+       and directly installs packages using conda.misc.explicit() as required by CEP-23.
+    2. For regular Environment instances: Uses the solver to determine the optimal
+       package set before installation.
+
+    :param prefix: The target installation path for the environment
+    :type prefix: str
+    :param specs: Package specifications to install
+    :type specs: list
+    :param args: Command-line arguments from the conda command
+    :type args: Namespace
+    :param env: Environment object containing dependencies and channels
+    :type env: Environment
+    :return: Installation result information
+    :rtype: dict
+
+    .. note::
+        This implementation follows CEP-23, which states: "When an explicit input file is
+        processed, the conda client SHOULD NOT invoke a solver."
+    """
+    # Handle explicit environments separately per CEP-23 requirements
+    if isinstance(env, ExplicitEnvironment):
+        return _install_explicit_environment(prefix, specs, env)
```

Comment:
This is a special case but one that will come up for any plugins that reads a definition for a locked environment (including pixi.lock or conda-lock.yml files).  In these cases a solve is unnecessary, should not be performed and in some cases (e.g. packages outside of a channel are referenced) may not be possible. I think providing a path for plugins to express this information would be useful but likely requires an expansion of the `Environment` class.

For pixi.lock and conda-lock.yml, parsing can return a set of `PackageRecord` instances for the conda requirements, where as requirement and environment.yaml files would return a set of `MatchSpec` instances.

Reply:
Another option would be to attaching this metadata to the `Environment` class itself which is returned from `EnvironmentSpecBase.environment` method. 

Either option is reasonable and will allow for plugins for various lock formats.

Reply:
Is there a need to get this feature ASAP? If not I'd prefer to wait until the Environment class is merged. Once that work is done this can be re-worked to use that class.